src/RefundsServiceProvider.php

```diff
@@ -19,7 +20,22 @@ public function configurePackage(Package $package): void
             ->name('refunds')
             ->hasConfigFile()
             ->hasViews()
-            ->hasMigration('create_refunds_table')
+            ->hasMigration('2020_05_07_100000_create_refunds_table')
             ->hasCommand(RefundsCommand::class);
     }
+
+    /**
+     * Using packageBooted lifecycle hooks to override the migration file name.
+     * We want to keep the old filename for now.
+     */
+    public function packageBooted()
+    {
+        foreach ($this->package->migrationFileNames as $migrationFileName) {
+            if (! $this->migrationFileExists($migrationFileName)) {
+                $this->publishes([
+                    $this->package->basePath("/../database/migrations/{$migrationFileName}.php.stub") => database_path('migrations/' . Str::finish($migrationFileName, '.php')),
+                ], "{$this->package->name}-migrations");
+            }
+        }
```

Comment:
I'm fine if we don't make the migration publishable. Let's talk about this in Slack and in our next Zoom meeting. In the other two packages, I opted to run the migrations directly from the package and not give the option to customize them:

- https://github.com/drewroberts/media
- https://github.com/drewroberts/blog

Reply:
CC: @sl0wik @pdbreen @joshtorres

Reply:
For the v1 (drop in replacement), I opted for checking if the table exists prior to publishing the core migrations.  With this, the existing migration file would remain within the tger/tger repo:

```
        if (! Schema::hasTable('discounts')) {
            $package->hasMigration('create_discounts_table');
        }
```

Reply:
The approach in this PR seems like it would only work with tger/tger repo.  For example, if I'm creating a new project and pull this package in, the migration will get created with a timestamp that's earlier than the migration that creates the users table it depends on.

Also not sure how this approach would work for future migrations within the package that add additional features.

Reply:
Let's discuss this some more today.